source/MRMesh/MRPointsComponents.cpp

```diff
@@ -0,0 +1,100 @@
+#include "MRPointsComponents.h"
+#include "MRPointCloud.h"
+#include "MRBitSet.h"
+#include "MRTimer.h"
+#include "MRPointsInBall.h"
+#include "MRProgressCallback.h"
+//#include <algorithm>
+
+namespace MR
+{
+
+namespace PointCloudComponents
+{
+
+Expected<std::vector<MR::VertBitSet>> getLargestComponents( const PointCloud& pointCloud, float maxDist, int minSize /*= -1*/, ProgressCallback pc /*= {}*/ )
+{
+    MR_TIMER
+
+    assert( maxDist > 0.f );
+    const auto& validPoints = pointCloud.validPoints;
+    if ( minSize < 0 )
+        minSize = std::max( int( validPoints.count() ) / 100, 1 );
```

Comment:
No need in this service. MinSize must be always specified by the user

Reply:
removed